cluster/swarm/cluster.go

```diff
@@ -176,7 +176,9 @@ func (c *Cluster) RemoveContainer(container *cluster.Container, force, volumes b
 
 // RemoveNetwork removes a network from the cluster
 func (c *Cluster) RemoveNetwork(network *cluster.Network) error {
-	return network.Engine.RemoveNetwork(network)
+	err := network.Engine.RemoveNetwork(network)
+	c.refreshNetworks()
```

Comment:
Also, don't we usually do that in the proxy code rather than here? (for instance with images)